diffkemp/cli.py

```diff
@@ -119,6 +119,9 @@ def make_argument_parser():
                             help="compare only selected function")
     compare_ap.add_argument("--custom-patterns", "-p",
                             help="custom pattern file or configuration")
+    compare_ap.add_argument("--use-smt",
+                            help="Use SMT-based checking of short snippets",
+                            action="store_true")
```

Comment:
Point for discussion: should we make the contents of this PR a new "builtin pattern" instead of a new CLI option? I can imagine that we'll eventually have more patterns which require SMT and we'd want to allow selecting just some. Using any such pattern would automatically turn on the SMT support.

Reply:
Interesting idea. I'd suggest keeping it as an option for now -- all patterns are enabled by default, right? Making this a pattern that's disabled by default feels a bit inconsistent. Furthermore, I perceive the builtin patterns as something rather lightweight and quite fast which definitely is not the case here. We can obviously change it later if we have more SMT-based patterns.

Reply:
> Interesting idea. I'd suggest keeping it as an option for now -- all patterns are enabled by default, right? Making this a pattern that's disabled by default feels a bit inconsistent. 

No, there are patterns which are off by default, namely `control-flow-only` and `type-casts`.

> Furthermore, I perceive the builtin patterns as something rather lightweight and quite fast which definitely is not the case here. We can obviously change it later if we have more SMT-based patterns.

Yes, but it feels like changing it later would remove the new `--use-smt` option (or at least change) which is not a nice breakage of the CLI. I'm a bit indecisive here, too, but I don't see any fundamental difference between this and built-in patterns. We'd just have to come up with a name for it.